pymongo_auth_aws/auth.py

```diff
@@ -154,6 +194,17 @@ def _aws_auth_header(credentials, server_nonce, sts_host):
     return final
 
 
+def _handle_credentials(func):
+    def inner(self, *args, **kwargs):
+        global _cached_credentials
+        try:
+            return func(self, *args, **kwargs)
+        except Exception as e:
```

Comment:
"as e" can be removed. Bare "raise" is preferable here to "raise e" because the former will retain the original traceback while the second will complicate it.

Reply:
Fixed